src/main/java/com/github/maleksandrowicz93/cqrsdemo/student/exception/StudentException.java

```diff
@@ -0,0 +1,15 @@
+package com.github.maleksandrowicz93.cqrsdemo.student.exception;
+
+import lombok.Getter;
+
+@Getter
+public abstract class StudentException extends Exception {
```

Comment:
nie musi być chyba public, a konstruktor wystarczy default (package)
nie jestem też przekonany co do tego, że jest to `Exception`, a nie `RuntimeException` - w tym przypadku tego nie czuć, bo wszystko opiera się na rzucaniu wyjątków wyżej i jednym handlerze, ale w bardziej skomplikowanym przypadku byłoby to API bardzo upierdliwe

Reply:
CheckedException mają taka wartość, cze są jakąś dokumentacja kodu, a w końcu i API modułu, czyli fasady. Wiemy, że możemy dostać na twarz pewne wyjątki. RuntimeException powinny być stosowane, kiedy użytkownik źle korzysta z API, np. pobiera 2 element z tablicy 1-elementowej, albo wywołuje toString na nullu. Wydaje mi się, że wyjątki tu zdefiniowane gdy są rzucane, nie są owocem złego korzystania z API, a jakąś formą outputu.

Reply:
W poprzednim komentarzu źle ustawiłem punkt skupienia - to nie checked/unchecked exception jest problemem, a API rzucające wyjątek w sytuacji "niezbyt wyjątkowej". Zawieranie logiki aplikacji w wyjątkach to przeciekanie abstrakcji tam, gdzie nie powinna. Od tego mamy API, aby prezentować rezultat działania w zwracanej wartości. Najprostszym rozwiązaniem byłby np. `Either` albo `Try` z vavra, ale nie jestem fanem tego rozwiązania, bo jest mało elastyczne i mniej czytelne względem po prostu zwracania value objectu zawierającego wszystkie informacje. 

Co konsument API z tym zrobi - nas to nie obchodzi co do zasady, ale jako że tutaj sam jesteś konsumentem API: możesz np. na podstawie obiektu rezultatu w kontrolerze zwrócić odpowiedź o błędzie w odpowiedni sposób. Korzyścią z tego podejścia jest dodatkowo to, że nie opieramy się na "magii springowej" typu global exception handler, kontroler przestaje być głupią przelotką HTTP, a faktycznie realizuje jakiś cel. Jest nim interpretacja wyniku wywołania metody i przygotowanie odpowiedniej odpowiedzi HTTP. Całość jest bardziej widoczna wprost E2E: wchodzisz do kontrolera -> wywołujesz logikę -> interpretujesz wynik. Bez skakania po dziwnych klasach i konieczności świadomości tego, że w Springu coś takiego jak global exception handler istnieje. Sztuką nie jest sama znajomość wszystkich zawiłości frameworków, a ich znajomość i odpowiednie (zwykle oszczędne) używanie ;) 

Co do checked exceptions:
- https://phauer.com/2015/checked-exceptions-are-evil/
- https://ohadshai.medium.com/its-almost-2020-and-yet-checked-exceptions-are-still-a-thing-bfaa24c8997e
- https://medium.com/javarevisited/its-2022-should-we-give-up-using-checked-exception-176429e05d4b

Reply:
Potencjalny refactor API ogarnę w kolejnej iteracji, najpierw chcę mieć pewność, że dla tego API projekt jest dobrze napisany. Musiałbym też chwilę pomyśleć jak takie API zorganizować, bo do tej pory obsługiwałem unhappy path tylko przez exception handlary głownie dlatego, że mogę rozdzielić zwrócenie poprawnej odpowiedzi od ErrorResponse z zachowaniem ścisłych typów w kontrolerze. No ale zgodnie z tym co mówisz, być może sytuacje typu pobranie nieistniejącego studenta nie powinny być "wyjątkowe". No, ale to na kolejną iterację :) NIemniej jednak, Exceptiony poprawione.